src/datautils/surfacedata.ts

```diff
@@ -177,53 +218,44 @@ const unassignedColorScale = scaleOrdinal<number, string>()
 /**
  * Find the best matching base index based on name or by values
  */
-function findBestMatchingBaseIndex(top: any, index: number, surfaces: any): number {
+function findBestMatchingBaseIndex(top: any, index: number, surfaces: any, stratColumn: any): number {
   const nextIndex: number = index + 1;
 
   if (!surfaces || nextIndex >= surfaces.length) {
     return null;
   }
 
-  // if there is a matching base by name, use that
-  let candidate: any = surfaces[nextIndex];
-  if (candidate.isBase && candidate.name === top.name) {
-    return nextIndex;
+  // If there is a matching base by name, use that. More robust, does not rely on sorting
+  const baseSurfaceIndex = surfaces.findIndex((candidate: any) => candidate.isBase && candidate.name === top.name);
+  if (baseSurfaceIndex !== -1) {
+    return baseSurfaceIndex;
   }
 
-  // Select nearest surface as base
-  const scores: any = [];
-  let ci: number = nextIndex;
-  for (; ci < surfaces.length; ci++) {
-    candidate = surfaces[ci];
-    let score: number = 0;
-    let count: number = 0;
-    for (let x = 0; x < top.values.length; x++) {
-      if (top.values[x] === null || candidate.values[x] === null) {
-        continue;
-      }
-      score += candidate.values[x] - top.values[x];
-      count++;
+  for (let i = nextIndex; i < surfaces.length; i++) {
+    const candidate = surfaces[i];
+    if (!candidate.isBase) {
+      return i;
     }
-    if (count) {
-      scores.push({ ci, score: score / count });
+    if (isAnchestor(top, candidate, stratColumn)) {
+      return i;
     }
   }
+  return null;
+}
 
-  if (scores.length < 2) {
-    return nextIndex;
-  }
-  scores.filter((s: any) => s.scores > 0).sort((a: any, b: any) => b.score - a.score || a.ci - b.ci);
-
-  return scores[0].ci;
+function isAnchestor(descendant: any, candidate: any, stratColumn: any): boolean {
```

Comment:
Lots of `any`'s. Don't we know the type?

Reply:
I'll look into it

Reply:
Replaced a number of anys where we actually have the type